pkg/kapp/diff/versioned_resource.go

```diff
@@ -91,6 +91,34 @@ func (d VersionedResource) updateAffected(rule ctlconf.TemplateRule, rs []ctlres
 		}
 	}
 
+	for _, res := range rs {
+		for k, v := range res.Annotations() {
+			if k == explicitReferenceKey || strings.HasPrefix(k, explicitReferenceKeyPrefix) {
+				explicitRef := NewExplicitVersionedRef(k, v)
+				objectRef, err := explicitRef.AsObjectRef()
+				if err != nil {
+					return fmt.Errorf("Parsing versioned explicit ref on resource '%s': %s", res.Description(), err)
+				}
+
+				// Passing empty TemplateAffectedObjRef as explicit references do not have a special name key
+				err = d.buildObjRefReplacementFunc(ctlconf.TemplateAffectedObjRef{})(objectRef)
+				if err != nil {
+					return fmt.Errorf("Parsing versioned explicit ref on resource '%s': %s", res.Description(), err)
```

Comment:
this isnt parsing...

Reply:
The only error thrown by these 2 functions are Marshalling/Unmarshalling errors. Doesn't it make sense to wrap those with something like `Parsing error` ?

Reply:
If we want to keep messages unique we could have 
```suggestion
					return fmt.Errorf("Creating explicit reference for resource  '%s': %s", res.Description(), err)
```